recipes/doxygen/all/conanfile.py

```diff
@@ -34,10 +34,15 @@ def configure(self):
         if minimum_compiler_version is not None:
             if tools.Version(self.settings.compiler.version) < minimum_compiler_version:
                 raise ConanInvalidConfiguration("Compiler version too old. At least {} is required.".format(minimum_compiler_version))
+        if (self.settings.compiler == "Visual Studio" and
+            tools.Version(self.settings.compiler.version.value) <= 14 and
+                tools.Version(self.version) == "1.8.18"):
+            raise ConanInvalidConfiguration("Doxygen version {} broken with VS {}.".format(self.version,
+                                                                                           self.settings.compiler.version))
```

Comment:
We don't know why only that version is failing..

Reply:
In one of the earlier runs I saw it mentioned that it failed running `flex` - I don't know how that would be specific to any particular version of Visual Studio though, unless it is some sort of race condition.

Reply:
I can reproduce the error on my machine when building doxygen 1.8.18 as:
```
conan create . doxygen/1.8.18@ -pr msvc14 -s build_type=Debug -s compiler.runtime=MTd
```
Building doxygen 1.8.18 with `build_type=Debug` and `runtime=MDd` also fails.

Building 1.9.1 with these configurations works, or switching to a newer compiler.

So I think you should only exclude Visual Studio 2015 with Debug runtime.
I did not check the root causes or what changes between the doxygen releases.

Reply:
I think the problem is the parallelization.
I find multiple instantiations of
```
C:\Users\maarten\.conan\data\winflexbison\2.5.22\_\_\package\f27dcd2d5e453cb01f9324a959461903ce8116a0\bin\win_flex.exe  -oC:/.conan/88ebf7/1/build_subfolder/source_subfolder/generated_src/configimpl.cpp configimpl.l
```
in the detailed build log.

By building with
```
cmake --build . --config Debug  -- /m:1
```
(instead of `/m:8`), I got it to build with Visual Studio 2015 Debug MTd.

So I propose to do something as follows:
```python
@property
def _can_build_parallel(self):
    if self.settings.compiler == "Visual Studio" and tools.Version(self.setting.compiler.version) <= 14 and "d" in str(self.settings.compiler.runtime):
        if self.version == "1.8.18":
            self.output.warn("Disabling parallel build due to multiple interfering invocations of win_flex")
            return False
    return True
def build(self):
    with tools.no_op() if self._can_build_parallel else tools.environment_append({"CONAN_CPU_COUNT": "1"}) :
        cmake.build()
```

Reply:
Sounds good.

Reply:
Also, CMake helper accepts parallel flag: https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#parallel. It can be simplified.